File: root/services/watersource.py
import json
import uuid
from Datamodel.Location import location

class Watersource:
    def __init__(self, location: location) -> None:
        self._id = str(uuid.uuid4())
        self._location = location

    def to_dict(self):
        return {
            "id": self._id,
            "location": self._location.to_dict()
        }
    
    def to_json(self):
        return json.dumps(self.to_dict())

File: tree-client/tree/Model/WaterSource.swift
//
//  WaterSource.swift
//  tree
//
//  Created by Nikolai Madlener on 18.11.23.
//

import Foundation

struct WaterSource: LocationObject, Decodable {
    var id: UUID
    var location: LatLon
    
    static func == (lhs: WaterSource, rhs: WaterSource) -> Bool {
        return lhs.id == rhs.id
    }
    
    // MARK: Codable methods.
    
    enum CodingKeys: String, CodingKey {
        case id
        case location
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.id = try container.decode(UUID.self, forKey: .id)
        self.location = try container.decode(LatLon.self, forKey: .location)
    }
    
    init(id: UUID, location: LatLon) {
        self.id = id
        self.location = location
    }
}


